File: app/src/main/java/com/magdenbt/collectionsbenchmark/di/mainactivity/MainModule.java
package com.magdenbt.collectionsbenchmark.di.mainactivity;

import androidx.fragment.app.FragmentActivity;

import com.magdenbt.collectionsbenchmark.ui.viewflow.MainActivity;
import com.magdenbt.collectionsbenchmark.di.SharedCollSizeViewModelModule;

import dagger.Binds;
import dagger.Module;

@Module(includes = SharedCollSizeViewModelModule.class)
public abstract class MainModule {

    @Binds
    public abstract FragmentActivity bindFragmentActivity(MainActivity activity);


}

File: app/src/main/java/com/magdenbt/collectionsbenchmark/di/mainactivity/MainModule.kt
package com.magdenbt.collectionsbenchmark.di.mainactivity

import com.magdenbt.collectionsbenchmark.di.SharedCollSizeViewModelModule
import dagger.Module

@Module(includes = [SharedCollSizeViewModelModule::class])
abstract class MainModule {
//    @Binds
//    abstract fun bindFragmentActivity(mainActivity: MainActivity):FragmentActivity
}
